src/Features/CSharp/Portable/MakeRefStruct/CSharpMakeRefStructCodeFixProvider.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.MakeRefStruct
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    internal class CSharpMakeRefStructCodeFixProvider : CodeFixProvider
+    {
+        // Error CS8345: Field or auto-implemented property cannot be of certain type unless it is an instance member of a ref struct.
+        private const string CS8345 = nameof(CS8345);
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+            => ImmutableArray.Create(CS8345);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var span = context.Span;
+
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var structDeclaration = FindContainingStruct(root, span);
+
+            // CS8345 could be triggered when struct is already marked with `ref` but a property is static
+            if (structDeclaration != null && !structDeclaration.Modifiers.Any(m => m.IsKind(SyntaxKind.RefKeyword)))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        CSharpFeaturesResources.Make_ref_struct,
+                        c => FixCodeAsync(context.Document, context.Span, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        private async Task<Document> FixCodeAsync(Document document, TextSpan span, CancellationToken c)
+        {
+            var root = await document.GetSyntaxRootAsync(c).ConfigureAwait(false);
+
+            var structDeclaration = FindContainingStruct(root, span);
```

Comment:
maybe a comment saying that this will always succeed because you're using the same span that worked in RegisterCodeFixesAsync

Reply:
Do we really need it? I haven't found neither additional null checks in the "fixing" code, nor special comments in similar cases (for example [this](https://github.com/dotnet/roslyn/blob/3bb73cf7f31cda72f057ae50ab6ce4b147efb115/src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs#L89) code).

Reply:
I would add the comment